servicebroker/lib/routes/binding.js

```diff
@@ -20,57 +20,64 @@ module.exports = function(app) {
     if (typeof(policyJSON) === "undefined") {
       res.status(400).json({ "description": messageUtil.getMessage("POLICY_REQUIRED") });
     } else {
-      models.binding.findAll({ where: { appId: appId } }).then(function(result) {
-        var length = result.length;
-        if (length === 0) {//no binding , so create one
-          models.sequelize.transaction().then(function(t) {
-            models.binding.create({
-              bindingId: bindingId,
-              appId: appId,
-              serviceInstanceId: serviceInstanceId,
-              timestamp: new Date().getTime(),
-              transaction: t
-            }).then(function(result) {
-              apiServerUtil.attachPolicy(appId, policyJSON, function(error, response) {
-                var statusCode = response.statusCode;
-                if (!error && (statusCode == 200 || statusCode == 201)) {
-                  t.commit();
-                  res.status(statusCode).json({});
+      models.sequelize.transaction().then(function(t) {
 
-                } else {
-                  logger.error("Fail to bind service when call api server, error=>" + error);
+        models.service_instance.findById(serviceInstanceId).then(function(result) {
```

Comment:
This seems a bit unnecessary since the create should fail if the service instance doesn't exist. Just trying to keep the number of queries to a minimum here.